src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java

```diff
@@ -51,6 +55,50 @@ public static int getIthArgument(final String[] args, final int i, final int def
             return def;
         }
     }
+    
+    /**
+     * Get the colony id from the ith argument (two integers separated by a pipe char)
+     * @param args the list of arguments
+     * @param i the index of the argument you want
+     * @param def the default value
+     * @return the colony id
+     */
+    public static int getColonyIdFromArg(final String[] args, final int i, final int def)
+    {
+    	if (args.length < i + 1)
+    	{
+    		return def;
+    	}
+    	else
+    	{
+    		String[] split = args[i].split(Pattern.quote("|"));
+    		if (split.length == 2)
+    		{
+    			Integer result1 = Ints.tryParse(split[1]);
+    			if (null != result1) return result1.intValue();
+    		}
+    		else if (split.length == 1)
+    		{
+    			Integer result = Ints.tryParse(split[0]);
+    			if (null != result) return result.intValue();
+    		}
+    	}
+    	return def;
```

Comment:
the intent seems bigger than necessary here

Reply:
I don't quite get what you mean?